plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskListSection.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Handles storage and retrieval of a task list section
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\OnboardingTasks;
+
+/**
+ * Task List section class.
+ */
+class TaskListSection {
+
+	/**
+	 * Title.
+	 *
+	 * @var string
+	 */
+	public $id = '';
+
+	/**
+	 * Title.
+	 *
+	 * @var string
+	 */
+	public $title = '';
+
+	/**
+	 * Description.
+	 *
+	 * @var string
+	 */
+	public $description = '';
+
+	/**
+	 * Image.
+	 *
+	 * @var string
+	 */
+	public $image = '';
+
+	/**
+	 * Tasks.
+	 *
+	 * @var array
+	 */
+	public $task_names = array();
+
+	/**
+	 * Parent task list.
+	 *
+	 * @var TaskList
+	 */
+	protected $task_list;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array         $data Task list data.
+	 * @param TaskList|null $task_list Parent task list.
+	 */
+	public function __construct( $data = array(), $task_list = null ) {
+		$defaults = array(
+			'id'          => '',
+			'title'       => '',
+			'description' => '',
+			'image'       => '',
+			'tasks'       => array(),
+		);
+
+		$data = wp_parse_args( $data, $defaults );
+
+		$this->task_list   = $task_list;
+		$this->id          = $data['id'];
+		$this->title       = $data['title'];
+		$this->description = $data['description'];
+		$this->image       = $data['image'];
+		$this->task_names  = $data['task_names'];
```

Comment:
I don't want to block this PR, but maybe we can make this update in the near future if you're not opposed: 

My gut feeling is that we should be passing in instances of tasks as opposed to names only and mapping them automatically.

```
$class = 'Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\\' . $task_name;
```

This means that all tasks have to be under our namespace and this is not very extensible or usable for 3PDs (or even our own plugins) that want to extend these lists.

Reply:
Yeah I agree, I think that makes sense.
I assume this change would pertain more so to the `add_task` method in `TaskList.php`.
Given for sections I only send the `id's` to the front-end and map them inside the sectioned TaskList component.